apps/src/code-studio/pd/foorm/editor/form/FoormFormSaveBar.jsx

```diff
@@ -121,15 +125,16 @@ class FoormFormSaveBar extends Component {
     return this.state.formName && this.state.formName.match('^[a-z0-9_]+$');
   };
 
-  saveNewForm = () => {
+  saveNewForm = (saveAsNewVersion = false) => {
     const newFormName = `${this.state.formCategory}/${this.state.formName}`;
     $.ajax({
       url: `/foorm/forms`,
       type: 'post',
       contentType: 'application/json',
       processData: false,
       data: JSON.stringify({
-        name: newFormName,
+        name: saveAsNewVersion ? this.props.formName : newFormName,
```

Comment:
This "just works"? No controller changes necessary?

Reply:
yep! it was already taking version from params and defaulting to 0 if undefined.